### DISCUSSION TITLE
Storing files on baremetal

### DISCUSSION THREAD

**Discussion by lukeclements** [nasa/fprime#2309] [Q&A] [answered]
Hi, 

I'm currently working with an Arduino microcontroller and I'm interested in utilizing the onboard SD card slot. I'd like to know if the functionality for file storage on baremetal exists, and if so, how do I use it?

**Comment by LeStarch** [comment, marked as answer]
We don't have functionality for SD cards at the moment.  Thus you'd probably need to implement that.  To do so, you'd need to implement the interface seen in `Os::File` and `Os::Directory` to work with SD cards and compile that in as your implementation for the `Os::File.`

**Reply by lukeclements** [reply]
Okay, thank you for the information!
